mantidproject/mantid
# Add ability to specify multiple attenuation .out files in Pearl config

**Description of work.**

The Pearl yaml config file in the ISIS powder diffraction scripts now supports multiple attenuation file paths for .out files via a new parameter attenuation_files. This parameter is a list of dictionaries with each dictionary containing a name and path key. One of the files can be used by specifying a further parameter called attenuation_file that has a value equal to the name of one of the "attenuation_files"
The .out files are produced by an external Fortran piece of software that the Pearl instrument scientists use. The Pearl configuration previously just supported a single attenuation file via a parameter called attenuation_file_path

**Report to:** Nick Funnell

**To test:**

The following steps run a focus on some Pearl input files that are read from the ISIS archive. The focus step makes use of the attenuation .out files:

Unzip the following zip file
[PearlFocusMultiOutFilesTest.zip](https://github.com/mantidproject/mantid/files/5334301/PearlFocusMultiOutFilesTest.zip)
Update the paths in the User_experiment.yaml file and the test.py script to be appropriate for your machine
Run the test.py script within Workbench and check it completes successfully. You'll need ISIS archive access enabled
Change the attenuation_file parameter in the Pearl.Focus call in test.py script so it contains a nonsense value eg XXXX
Observe the script fails with an appropriate error
Change the attenuation_file parameter in the Pearl.Focus call in test.py script so it contains the value WC (which is the name of the other valid .out file in the .yaml config file)
Observe the script succeeds

Fixes #29422 .

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
